StreamChatClient_v3/WebSocketClient/WebSocketClient.swift

```diff
@@ -94,6 +82,15 @@ class WebSocketClient {
     /// An object containing external dependencies of `WebSocketClient`
     private let environment: Environment
     
+    private(set) lazy var pingController: WebSocketPingController =
+        environment.createPingController(environment.timerType,
+                                         engineQueue,
+                                         { [weak self] in self?.engine.sendPing() },
+                                         { [weak self] in
+                                             self?.disconnect()
```

Comment:
I think we need to add a new disconnection reason - something like `noPongReceived`. This call uses the default value, which is `userInitiated`. That's actually all you had to do I think. The `reconnectIfNeeded` refactoring is redundant. The original implementation of `websocketDidDisconnect` would handle this correctly and start reconnecting automatically. It seems all we need to do here is actually just to call disconnect with the correct reason :magic: